File: Java/Teste.java
package pilha;

public class Teste {

    public static void main(String[] args) {

        // Criando a pilha com capacidade 3
        Pilha minhaPilha = new Pilha(3);

        // Testando o método isEmpty
        System.out.println("A pilha está vazia? " + minhaPilha.isEmpty()); // Deve retornar true

        System.out.println("Empilhando elementos...");
        // Tentando empilhar um elemento quando a pilha estiver cheia
        try {
        	minhaPilha.push(10); // Empilha o número 10
            minhaPilha.push(20); // Empilha o número 20
            minhaPilha.push(30); // Empilha o número 30
            minhaPilha.push(40); // Deve lançar a exceção "Pilha cheia!"
        } catch (Exception e) {
            System.out.println("Erro: " + e.getMessage());
        }

        // Verificando o topo da pilha
        try {
			System.out.println("\nTopo da pilha: " + minhaPilha.topo()); // Deve imprimir 30
		} catch (Exception e) {
            System.out.println("Erro: " + e.getMessage());
		}

        // Testando desempilhamento
        System.out.println("\nDesempilhando elementos...");
        // Tentando desempilhar da pilha vazia
        try {
        	minhaPilha.pop(); // Deve imprimir 30
            minhaPilha.pop(); // Deve imprimir 20
            minhaPilha.pop(); // Deve imprimir 10
            minhaPilha.pop(); // Deve lançar a exceção "Pilha vazia!"
        } catch (Exception e) {
            System.out.println("Erro: " + e.getMessage());
        }

        // Testando novamente o método isEmpty
        System.out.println("A pilha está vazia? " + minhaPilha.isEmpty()); // Deve retornar true
    }
}

File: C/teste.c
#include <stdio.h>
#include <stdlib.h>
#include "pilha.h"

int main()
{
    Pilha *minha_pilha = criar_pilha(5);
    if (minha_pilha == NULL)
    {
        return -1;
    }

    bool esta_vazio = is_empty(minha_pilha);
    printf("A pilha esta vazia? %s\n", esta_vazio ? "true" : "false");

    printf("Empilhando elementos...\n");
    push(10, minha_pilha);  // Empilha o número 10
    push(20, minha_pilha);  // Empilha o número 20
    push(30, minha_pilha);  // Empilha o número 30
    push(40, minha_pilha);  // Empilha o número 40
    push(50, minha_pilha);  // Empilha o número 50
    push(60, minha_pilha);  // Tenta empilhar mais um, pilha cheia

    printf("\nTopo da pilha: ");
    topo(minha_pilha);

    printf("\nDesempilhando elementos...\n");
    pop(minha_pilha);  // Desempilha o topo
    pop(minha_pilha);  // Desempilha o topo

    printf("\nTopo da pilha: ");
    topo(minha_pilha); // Verifica o topo após desempilhar dois elementos

    printf("\nDesempilhando ate a pilha ficar vazia...\n");
    pop(minha_pilha);  // Desempilha o topo
    pop(minha_pilha);  // Desempilha o topo
    pop(minha_pilha);  // Desempilha o topo
    pop(minha_pilha);  // Desempilha o topo

    printf("\nTopo da pilha: ");
    topo(minha_pilha);
    esta_vazio = is_empty(minha_pilha);
    printf("A pilha esta vazia? %s\n", esta_vazio ? "true" : "false");

    // Liberando a memória
    liberar_pilha(minha_pilha);

    return 0;
}
